include/deal.II/arborx/bvh.h

```diff
@@ -0,0 +1,106 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2021 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_arborx_bvh_h
+#define dealii_arborx_bvh_h
+
+#include <deal.II/base/config.h>
+
+#if defined(DEAL_II_WITH_ARBORX) && defined(DEAL_II_HAVE_CXX17)
+#  include <ArborX_LinearBVH.hpp>
+#  include <Kokkos_Core.hpp>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * This class implements a wrapper around ArborX::BVH. BVH stands from Bounding
+ * Volume Hierarchy.
+ *
+ * From [Wikipedia](https://en.wikipedia.org/wiki/Bounding_Volume_Hierarchy):
+ * <blockquote>
+ * A bounding volume hierarchy (BVH) is a tree structure on a set of geometric
+ * objects. All geometric objects are wrapped in bounding volumes that form the
+ * leaf nodes of the tree. These nodes are then grouped as small sets and
+ * enclosed within larger bounding volumes. These, in turn, are also grouped and
+ * enclosed within other larger bounding volumes in a recursive fashion,
+ * eventually resulting in a tree structure with a single bounding volume at the
+ * top of the tree. Bounding volume hierarchies are used to support several
+ * operations on sets of geometric objects efficiently, such as in collision
+ * detection and ray tracing.
+ * </blockquote>
+ *
+ * Because ArborX uses Kokkos, Kokkos needs to be initialized and finalized.
+ * This can be done using Utilities::MPI::MPI_InitFinalize.
```

Comment:
Hmm... This seems to misuse `Utilities::MPI::MPI_InitFinalize` especially when compiling without MPI support. I would rather let the user do it explicitly.